display_list/skia/dl_sk_paint_dispatcher_unittests.cc

```diff
@@ -31,5 +43,44 @@ TEST(DisplayListUtils, OverRestore) {
   helper.restore();
 }
 
+// https://github.com/flutter/flutter/issues/132860.
+TEST(DisplayListUtils, SetDitherIgnoredIfColorSourceNotGradient) {
+  MockDispatchHelper helper;
+  helper.setDither(true);
+  EXPECT_FALSE(helper.paint().isDither());
+}
+
+// https://github.com/flutter/flutter/issues/132860.
+TEST(DisplayListUtils, SetColorSourceClearsDitherIfNotGradient) {
+  MockDispatchHelper helper;
+  helper.setDither(true);
+  helper.setColorSource(nullptr);
+  EXPECT_FALSE(helper.paint().isDither());
+}
+
+// https://github.com/flutter/flutter/issues/132860.
+TEST(DisplayListUtils, SetDitherTrueThenSetColorSourceDithersIfGradient) {
+  MockDispatchHelper helper;
+
+  // A naive implementation would ignore the dither flag here since the current
+  // color source is not a gradient.
+  helper.setDither(true);
+  helper.setColorSource(kTestLinearGradient.get());
+  EXPECT_TRUE(helper.paint().isDither());
+}
+
+// https://github.com/flutter/flutter/issues/132860.
+TEST(DisplayListUtils, SetDitherTrueThenRenderNonGradientThenRenderGradient) {
+  MockDispatchHelper helper;
+
+  // "Render" a dithered non-gradient.
+  helper.setDither(true);
+  EXPECT_FALSE(helper.paint().isDither());
+
+  // "Render" a gradient.
+  helper.setColorSource(kTestLinearGradient.get());
+  EXPECT_TRUE(helper.paint().isDither());
+}
```

Comment:
Add a test that does:

dither=true
source=gradient
(true)
source=non-gradient
(false)
source=gradient
(true)
dither=false
(false)

Reply:
Added!